ImperatorToCk2/src/main/java/ImperatorToCK2/Importer.java

```diff
@@ -1326,5 +1329,364 @@ public static String[] importSaveInfo (String directory) throws IOException //im
 
     }
 
+    public static ArrayList<String[]> importFlag (String name) throws IOException
+    {
+        String VM = "\\";
+        VM = VM.substring(0);
+        name = name+VM+"game"+VM+"common"+VM+"coat_of_arms"+VM+"coat_of_arms"+VM+"00_pre_scripted_countries.txt";
+        
+        FileInputStream fileIn= new FileInputStream(name);
+        Scanner scnr= new Scanner(fileIn);
+
+        ArrayList<String[]> impFlagList= new ArrayList<String[]>();
+
+        String tab = "	";
+        char quote = '"';
+        String strQuote = quote+"_";
+        strQuote = strQuote.split("_")[0];
+        int flag = 0;
+
+        boolean endOrNot = true;
+        String qaaa;
+        String[] output;   // Owner Culture Religeon PopTotal Buildings
+        output = new String[5];
+
+        output[0] = "unnamedFlag"; //default for no tag/name
+        output[1] = "pattern_solid.tgaq"; //default for no pattern
+        output[2] = "none"; //default for color1, format = hsvOrRgb,r g b
+        output[3] = "none"; //default for color2, format = hsvOrRgb,r g b
+        output[4] = "0"; //default for no emblems, format is texture~_~color1~_~color2~_~scale~_~position~_~rotation~~(nextEmblem)
+        //~_~ divides aspects of an emblem, ~~ divides emblemst
+
+        impFlagList.add(output); //default at ID 0
+
+        try {
+            while (endOrNot = true){
+                endOrNot = false;
+
+                while (flag == 0) {
```

Comment:
Looks like you never set flag, so could remove the variable and the loop control here.